tests/cache/test_memory_backend.py

```diff
@@ -10,5 +17,30 @@ class MemoryBackendSerializerTest(_GenericSerializerTest, MemoryBackendTest):
     pass
 
 
+def raise_cant_deserialize_exception(v):
+    raise CantDeserializeException()
+
+
+class MemoryBackendSerializerCantDeserializeExceptionTest(
+    _GenericBackendFixture, TestCase
+):
+    backend = "dogpile.cache.memory"
+
+    region_args = {
+        "serializer": lambda v: json.dumps(v).encode("ascii"),
+        "deserializer": raise_cant_deserialize_exception,
+    }
+
+    def test_serializer_cant_deserialize(self):
+        region = self._region()
```

Comment:
**mike bayer** (zzzeek) wrote:

this test is fine but feels like it should be part of the generic suite.

We could for example add this test to the existing _fixtures._GenericSerializerTest like this:

	diff --git a/tests/cache/_fixtures.py b/tests/cache/_fixtures.py
	index 9e71f8f..60ee281 100644
	--- a/tests/cache/_fixtures.py
	+++ b/tests/cache/_fixtures.py
	@@ -388,6 +388,18 @@ class _GenericSerializerTest(TestCase):
	         "deserializer": json.loads,
	     }
	 
	+    def test_serializer_cant_deserialize(self):
	+        region = self._region(region_args={
	+            "serializer": self.region_args["serializer"],
	+            "deserializer":raise_cant_deserialize_exception,
	+        })
	+
	+        value = {"foo": ["bar", 1, False, None]}
	+        region.set("k", value)
	+        asserted = region.get("k")
	+        eq_(asserted, NO_VALUE)
	+        
	+
	     def test_uses_serializer(self):
	         region = self._region()


that way it gets run for all the backends that include a _GenericSerializerTest class, which is all of them (memory, redis, memcached, dbm).

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/4370

Reply:
yeah, I didn't like it being in the memory serializer. good idea for the change. I'll implement it